libraries/chain/db_maint.cpp

```diff
@@ -780,6 +780,16 @@ double database::calculate_vesting_factor(const account_object& stake_account)
    uint32_t current_subperiod = get_gpos_current_subperiod();
  
    if(current_subperiod == 0 || current_subperiod > number_of_subperiods) return 0;
+
+   // On starting new vesting period, all votes become zero until some one votes, To avoid a situation of zero votes, 
+   // changes done to roll in GPOS rules, the vesting factor will be 1 for who ever votes in 6th sub-period of last vesting period
```

Comment:
English is not my native language but it seems the text haves some problems. Maybe:

On starting A new vesting period, all votes become zero until SOMEONE votes. To avoid a situation of zero votes, changes WERE done to roll in GPOS rules, the vesting factor will be 1 for WHOEVER votes in 6th sub-period of last vesting period.

By the way, what are the changes done to GPOS rules you refer to ?

Reply:
GPOS rules in the sense that awarding vesting factor 1 without voting(based on voting status of 6th sub-period of last vesting period). I feel it is nothing but tweaking GPOS rules. 
If required, I can update the comments.